test/Helsenorge.Messaging.Tests/Mocks/TestCertificates.cs

```diff
@@ -18,56 +20,15 @@ namespace Helsenorge.Messaging.Tests.Mocks
-
-        public static X509Certificate2 HelsenorgePublicSignature => new X509Certificate2(TestFileUtility.GetFullPathToFile($"Files{Path.DirectorySeparatorChar}Helsenorge_Signature.cer"));
-
-        public const string HelsenorgeEncryptionThumbprint = "fddbcfbb3231f0c66ee2168358229d3cac95e88a";
-        public static X509Certificate2 HelsenorgePrivateEncryption => new X509Certificate2(TestFileUtility.GetFullPathToFile($"Files{Path.DirectorySeparatorChar}Helsenorge_Encryption.p12"), HelsenorgeCertificatePassword.ToSecureString());
-        
-        public const string HelsenorgeEncryptionInvalidStartThumbPrint = "be26ea7a3eb54e7b00c55782304765777f854192";
-        //makecert -r -pe -b 01/01/2020 -e 01/01/2030 -n "CN=XYZ Encryption start" -ss my -sky 1 Helsenorge_EncryptionInvalidStart.cer
-        public static X509Certificate2 HelsenorgePrivateEncryptionInvalidStart => new X509Certificate2(TestFileUtility.GetFullPathToFile($"Files{Path.DirectorySeparatorChar}Helsenorge_EncryptionInvalidStart.pfx"), HelsenorgeCertificatePassword.ToSecureString());
-
-        public const string HelsenorgeEncryptionInvalidEndThumbprint = "be26ea7a3eb54e7b00c55782304765777f854191";
-        // makecert -r -pe -b 01/01/2005 -e 01/01/2010 -n "CN=XYZ Encryption end" -ss my -sky 1 Helsenorge_EncryptionInvalidEnd.cer
-        public static X509Certificate2 HelsenorgePrivateEncryptionInvalidEnd => new X509Certificate2(TestFileUtility.GetFullPathToFile($"Files{Path.DirectorySeparatorChar}Helsenorge_EncryptionInvalidEnd.pfx"), HelsenorgeCertificatePassword.ToSecureString());
-
-        public const string HelsenorgeSigntatureThumbprint = "bd302b20fcdcf3766bf0bcba485dfb4b2bfe1379";
-        public static X509Certificate2 HelsenorgePrivateSigntature => new X509Certificate2(TestFileUtility.GetFullPathToFile($"Files{Path.DirectorySeparatorChar}Helsenorge_Signature.p12"), HelsenorgeCertificatePassword.ToSecureString());
-
-        private static SecureString ToSecureString(this string source)
+        public static X509Certificate2 GenerateX509Certificate2 (X509KeyUsageFlags x509KeyUsage,DateTimeOffset notBefore, DateTimeOffset notAfter)
         {
-            if (string.IsNullOrWhiteSpace(source))
-                return null;
-            else
-            {
-                var result = new SecureString();
-                foreach (var c in source.ToCharArray())
-                    result.AppendChar(c);
-                return result;
-            }
+            using RSA rsa = RSA.Create(2048);
+            var req = new CertificateRequest("cn=HelseNorge", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+            req.CertificateExtensions.Add(new X509KeyUsageExtension(x509KeyUsage, true));
+            var cert = req.CreateSelfSigned(notBefore, notAfter);
+            return cert;
```

Comment:
Please spell out the variable names like:
- `certificateRequest` instead of `req`
- `selfSignedCertificate instead of `cert`